agent/config/builder.go

```diff
@@ -1113,12 +1132,50 @@ func (b *Builder) Validate(rt RuntimeConfig) error {
 		return fmt.Errorf("data_dir cannot be empty")
 	}
 
-	if !validContentPath.MatchString(rt.UIContentPath) {
-		return fmt.Errorf("ui-content-path can only contain alphanumeric, -, _, or /. received: %s", rt.UIContentPath)
+	if !validContentPath.MatchString(rt.UIConfig.ContentPath) {
+		return fmt.Errorf("ui-content-path can only contain alphanumeric, -, _, or /. received: %q", rt.UIConfig.ContentPath)
+	}
+
+	if hasVersion.MatchString(rt.UIConfig.ContentPath) {
+		return fmt.Errorf("ui-content-path cannot have 'v[0-9]'. received: %q", rt.UIConfig.ContentPath)
 	}
 
-	if hasVersion.MatchString(rt.UIContentPath) {
-		return fmt.Errorf("ui-content-path cannot have 'v[0-9]'. received: %s", rt.UIContentPath)
+	if rt.UIConfig.MetricsProvider != "" &&
+		!reBasicName.MatchString(rt.UIConfig.MetricsProvider) {
+		return fmt.Errorf("ui_config.metrics_provider can only contain lowercase "+
+			"alphanumeric or _ characters. received: %q", rt.UIConfig.MetricsProvider)
+	}
+	if rt.UIConfig.MetricsProviderOptionsJSON != "" {
+		// Attempt to parse the JSON to ensure it's valid, parsing into a map
+		// ensures we get an object.
+		var dummyMap map[string]interface{}
+		err := json.Unmarshal([]byte(rt.UIConfig.MetricsProviderOptionsJSON),
+			&dummyMap)
```

Comment:
Minor: kind of awkward line wrapping (https://github.com/golang/go/wiki/CodeReviewComments#line-length)

Reply:
I'd tend to agree with you. When I started here it was tribal knowledge that there is a preference in our projects to limit lines to 100 chars max and I was asked to stick to this in several PRs.

I suggested at the time that if this is important we codify it as it's not a general Go community thing but that document  still doesn't exist. I don't know how widespread the practice is but since I setup those line length guides in my editor I still use them.

I'd be happy if the consensus has moved on from that - I know it's not rigorously adhered to anyway - but at one point this was a specific thing people ere seemed to care about the other way 🤷 .

Reply:
TBF this one was super border line anyway so I just fixed it - obviously more readable on one line. 😄